File: ResourceManager/Models/Article.cs
using ResourceManager.Models.Abstractions;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace ResourceManager.Models
{
    public class Article : BaseEntity, ITags, IEquatable<Article>, IComparable<Article>
    {
        public string Title { get; set; } = string.Empty;
        public string Author { get; set; } = string.Empty;
        public string Filename { get; set; } = string.Empty;
        public string FilePath
        {
            get
            {
                if (!string.IsNullOrEmpty(Filename))
                {
                    return Path.Combine(App.Settings.ArticleSettings.FolderPath, Filename);
                }
                return Filename;
            }
        }
        public override string GetCaption()
        {
            return string.Format("{0} with title '{1}'", nameof(Article), Title);
        }
        public IList<ArticleTag> ArticleTags { get; set; } = new List<ArticleTag>();

        public string TagString => string.Join(",", ArticleTags.Select(x => x.Tag.Name));
        #region IEquatable<Article>
        public bool Equals(Article? other)
        {
            if (ReferenceEquals(null, other)) return false;

            return Title.Equals(other.Title);
        }
        #endregion

        #region IComparable<Article>
        public int CompareTo(Article? other)
        {
            if (ReferenceEquals(null, other)) return -1;

            return Title.CompareTo(other.Title);
        }
        #endregion
    }
}

File: ResourceManager/Models/Article.vb
Imports ResourceManager.Models.Abstractions
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq

Namespace ResourceManager.Models
    Public Class Article
        Inherits BaseEntity
        Implements ITags, IEquatable(Of Article), IComparable(Of Article)
        Public Property Title As String = String.Empty
        Public Property Author As String = String.Empty
        Public Property Filename As String = String.Empty
        Public ReadOnly Property FilePath As String
            Get
                If Not String.IsNullOrEmpty(Filename) Then
                    Return Path.Combine(Application.Settings.ArticleSettings.FolderPath, Filename)
                End If
                Return Filename
            End Get
        End Property
        Public Overrides Function GetCaption() As String
            Return String.Format("{0} with title '{1}'", NameOf(Article), Title)
        End Function
        Public Property ArticleTags As ICollection(Of ArticleTag)

        Public ReadOnly Property TagString As String Implements ITags.TagString
            Get
                Return String.Join(",", ArticleTags.Select(Function(x) x.Tag.Name))
            End Get
        End Property
#Region "IEquatable<Article>"
        Public Shadows Function Equals(ByVal other As Article) As Boolean Implements IEquatable(Of Article).Equals
            If ReferenceEquals(Nothing, other) Then Return False

            Return Title.Equals(other.Title)
        End Function
#End Region

#Region "IComparable<Article>"
        Public Function CompareTo(ByVal other As Article) As Integer Implements IComparable(Of Article).CompareTo
            If ReferenceEquals(Nothing, other) Then Return -1

            Return Title.CompareTo(other.Title)
        End Function
#End Region
    End Class
End Namespace
